ReactNativeClient/lib/services/KeymapService.ts

```diff
@@ -0,0 +1,317 @@
+			return item.accelerator;
+		}
+	}
+
+	setAccelerator(command: string, accelerator: string) {
+		// Run through localizeAccelerator() to detect duplicates with high accuracy
+		this.keymap[command].accelerator = KeymapService.localizeAccelerator(accelerator);
+	}
+
+	resetAccelerator(command: string) {
+		const defaultItem = KeymapService.defaultKeymap.find((item => item.command === command));
+		this.setAccelerator(command, defaultItem.accelerator);
+	}
+
+	getKeymap() {
+		const defaultItemsByCommand = KeymapService.initKeymap();
+
+		// Compare each "accelerator" value of the current keymap with the default values
+		// Return a keymap object, which can later be converted to a string and saved to the disk
+		return Object.entries(this.keymap).reduce((customKeymap, [command, { accelerator }]) => {
+			// If the Accelerator has been changed, add into the customKeymap object
+			if (defaultItemsByCommand[command].accelerator !== accelerator) {
+				customKeymap.push({ command, accelerator });
+			}
+			return customKeymap;
+		}, []);
+	}
+
+	setKeymap(customKeymap: KeymapItem[]) {
+		this.logger().info('KeymapService: Overriding the keymap...');
+
+		for (let i = 0; i < customKeymap.length; i++) {
+			const item = customKeymap[i];
+
+			try {
+				this.validateKeymapItem(item); // Throws if there are any issues in the keymap item
+				this.setAccelerator(item.command, item.accelerator);
+			} catch (err) {
+				throw new Error(
+					`Keymap item ${JSON.stringify(item)} is invalid!\nReason: ${err.message}`
+				);
+			}
+		}
+
+		try {
+			this.validateKeymap(); // Throws whenever there are duplicate Accelerators used in the keymap
+		} catch (err) {
+			throw new Error(
+				`Keymap configuration contains duplicates!\n${err.message}`
+			);
+		}
+	}
+
+	resetKeymap() {
+		this.keymap = KeymapService.initKeymap();
+	}
+
+	validateKeymapItem(item: KeymapItem) {
+		const commandReference = 'https://github.com/laurent22/joplin/blob/master/README.md';
+		const acceleratorReference = 'https://www.electronjs.org/docs/api/accelerator';
+
+		if (!item.hasOwnProperty('command')) {
+			throw new Error(
+				'Couldn\'t find the "command" property.\n\n' +
+				`Visit ${commandReference} to see the list of available commands.`
+			);
+		} else if (!this.keymap.hasOwnProperty(item.command)) {
+			throw new Error(
+				`"${item.command}" is not a valid command.\n\n` +
+				`Visit ${commandReference} to see the list of available commands.`
+			);
+		}
+
+		if (!item.hasOwnProperty('accelerator')) {
+			throw new Error(
+				'Couldn\'t find the "accelerator" property.\n\n' +
+				`Visit ${acceleratorReference} to learn more about Electron Accelerators.`
+			);
+		} else if (!(item.accelerator === null || KeymapService.isAccelerator(item.accelerator))) {
+			throw new Error(
+				`"${item.accelerator}" is not a valid Accelerator.\n\n` +
+				`Visit ${acceleratorReference} to learn more about Electron Accelerators.`
+			);
+		}
+	}
+
+	validateKeymap() {
```

Comment:
I don't see this method being referenced outside of the class. If that's indeed the case, please mark it as private. In general, please mark all non-public methods as private.